timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/PaxosTimeLockServerIntegrationTest.java

```diff
@@ -188,228 +156,121 @@ public void lockServiceShouldNotAllowUsToRefreshLocksFromDifferentNamespaces() t
 
-        LockToken token1 = lockService1.lock(newLockV2Request(LOCK_1)).getToken();
-        LockToken token2 = lockService2.lock(newLockV2Request(LOCK_1)).getToken();
-
-        lockService1.unlock(ImmutableSet.of(token1));
-        lockService2.unlock(ImmutableSet.of(token2));
+        namespace1.unlock(token1);
+        namespace2.unlock(token2);
     }
 
     @Test
-    public void asyncLockServiceShouldNotAllowUsToRefreshLocksFromDifferentNamespaces() throws InterruptedException {
-        TimelockService lockService1 = getTimelockService(CLIENT_1);
-        TimelockService lockService2 = getTimelockService(CLIENT_2);
-
-        LockToken token = lockService1.lock(newLockV2Request(LOCK_1)).getToken();
+    public void asyncLockServiceShouldNotAllowUsToRefreshLocksFromDifferentNamespaces() {
+        LockToken token = namespace1.lock(newLockV2Request(LOCK_1)).getToken();
 
-        assertThat(lockService1.refreshLockLeases(ImmutableSet.of(token))).isNotEmpty();
-        assertThat(lockService2.refreshLockLeases(ImmutableSet.of(token))).isEmpty();
+        assertThat(namespace1.refreshLockLease(token)).isTrue();
+        assertThat(namespace2.refreshLockLease(token)).isFalse();
 
-        lockService1.unlock(ImmutableSet.of(token));
+        namespace1.unlock(token);
     }
 
     @Test
     public void timestampServiceShouldGiveUsIncrementalTimestamps() {
-        long timestamp1 = timestampService.getFreshTimestamp();
-        long timestamp2 = timestampService.getFreshTimestamp();
+        long timestamp1 = namespace1.getFreshTimestamp();
+        long timestamp2 = namespace1.getFreshTimestamp();
 
         assertThat(timestamp1).isLessThan(timestamp2);
     }
 
     @Test
     public void timestampServiceShouldRespectDistinctClientsWhenIssuingTimestamps() {
-        TimestampService timestampService1 = getTimestampService(CLIENT_1);
-        TimestampService timestampService2 = getTimestampService(CLIENT_2);
-
-        long firstServiceFirstTimestamp = timestampService1.getFreshTimestamp();
-        long secondServiceFirstTimestamp = timestampService2.getFreshTimestamp();
+        long firstServiceFirstTimestamp = namespace1.getFreshTimestamp();
+        long secondServiceFirstTimestamp = namespace2.getFreshTimestamp();
 
-        getFortyTwoFreshTimestamps(timestampService1);
+        getFortyTwoFreshTimestamps(namespace1.timelockService());
 
-        long firstServiceSecondTimestamp = timestampService1.getFreshTimestamp();
-        long secondServiceSecondTimestamp = timestampService2.getFreshTimestamp();
+        long firstServiceSecondTimestamp = namespace1.getFreshTimestamp();
+        long secondServiceSecondTimestamp = namespace2.getFreshTimestamp();
 
         assertThat(firstServiceSecondTimestamp - firstServiceFirstTimestamp).isGreaterThanOrEqualTo(FORTY_TWO);
         assertThat(secondServiceSecondTimestamp - secondServiceFirstTimestamp).isBetween(0L, (long) FORTY_TWO);
```

Comment:
non-actionable: sigh, in practice this looks like a flake waiting to happen

Reply:
hasn't flaked so far!!